tests/errmsgs/treportunused.nim

```diff
@@ -4,13 +4,13 @@ treportunused.nim(19, 10) Hint: 'treportunused.s1(a: string)[declared in treport
 treportunused.nim(26, 5) Hint: 's8' is declared but not used [XDeclaredButNotUsed]
 treportunused.nim(22, 6) Hint: 'treportunused.s4()[declared in treportunused.nim(22, 5)]' is declared but not used [XDeclaredButNotUsed]
 treportunused.nim(25, 7) Hint: 's7' is declared but not used [XDeclaredButNotUsed]
-treportunused.nim(24, 7) Hint: 's6' is declared but not used [XDeclaredButNotUsed]
+treportunused.nim(24, 7) Hint: 'treportunused.s6(a: int)[declared in treportunused.nim(24, 6)]' is declared but not used [XDeclaredButNotUsed]
 treportunused.nim(23, 6) Hint: 'treportunused.s5(a: T)[declared in treportunused.nim(23, 5)]' is declared but not used [XDeclaredButNotUsed]
 treportunused.nim(20, 10) Hint: 'treportunused.s2()[declared in treportunused.nim(20, 9)]' is declared but not used [XDeclaredButNotUsed]
-treportunused.nim(29, 6) Hint: 's11' is declared but not used [XDeclaredButNotUsed]
+treportunused.nim(29, 6) Hint: 'treportunused.s11[declared in treportunused.nim(29, 5)]' is declared but not used [XDeclaredButNotUsed]
 treportunused.nim(27, 5) Hint: 's9' is declared but not used [XDeclaredButNotUsed]
-treportunused.nim(21, 10) Hint: 's3' is declared but not used [XDeclaredButNotUsed]
-treportunused.nim(28, 6) Hint: 's10' is declared but not used [XDeclaredButNotUsed]
+treportunused.nim(21, 10) Hint: 'treportunused.s3()[declared in treportunused.nim(21, 9)]' is declared but not used [XDeclaredButNotUsed]
```

Comment:
don't you think that this is a bit redundant? Print line information twice?

Reply:
yes, but that's a separate, pre-existing issue: look at this file in devel: https://github.com/nim-lang/Nim/blob/f18fcf65b3a61b1960c14aa9503eea34fee76492/tests/errmsgs/treportunused.nim#L3
```
discard """
  nimout: '''
treportunused.nim(19, 10) Hint: 'treportunused.s1(a: string)[declared in treportunused.nim(19, 9)]' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(26, 5) Hint: 's8' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(22, 6) Hint: 'treportunused.s4()[declared in treportunused.nim(22, 5)]' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(25, 7) Hint: 's7' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(24, 7) Hint: 's6' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(23, 6) Hint: 'treportunused.s5(a: T)[declared in treportunused.nim(23, 5)]' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(20, 10) Hint: 'treportunused.s2()[declared in treportunused.nim(20, 9)]' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(29, 6) Hint: 's11' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(27, 5) Hint: 's9' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(21, 10) Hint: 's3' is declared but not used [XDeclaredButNotUsed]
treportunused.nim(28, 6) Hint: 's10' is declared but not used [XDeclaredButNotUsed]
'''
```
it already contains the redundant info, but only for some kinds (eg iterator) and not some other kinds (eg macro), which is arbitrary.

I'm happy to file an issue for this pre-existing issue, but it's a minor annoyance that (IIUC) should only happen with `XDeclaredButNotUsed` ; other error messages will benefit from the extra info as it won't be redundant in these other cases.

Reply:
I don't see those "other error messages [that] will benefit from the extra info".

Reply:
reported that separate issue regarding `XDeclaredButNotUsed` being redundant here: https://github.com/nim-lang/Nim/issues/10101

> I don't see those "other error messages [that] will benefit from the extra info".

 see https://github.com/nim-lang/Nim/pull/9766#issuecomment-449714771 that shows how #8794 is fixed; I can add other tests that show how the other issues are being fixed; but AFAIL this particular comment is addressed, marking as resolved